### ISSUE TITLE
Reconstruction issue

### ISSUE THREAD

**Issue by starflame-Qot** [STFS-TUDa/blastAMR#7]
hello，I tested two tutorial cases, freeProFlame2D and damBreak2D, and both worked fine in reconstructParMesh. 

However, once you use reconstructPar, I will get the following error, as shown in the figure. 

![Screenshot from 2023-12-22 08-43-27](https://github.com/STFS-TUDa/blastAMR/assets/119994097/9b91f70b-9a4b-429e-8441-2c09bbc89979)

It doesn't go wrong from the beginning, it happens sometime in the middle. When I use _reconstructPar -latestTime_, I can synthesize the data for the last time.

Thanks for your potential help.

**Comment by JanGaertner** [comment]
I had the same issue, never could figure out why it happens as it appears kinda random. 

However, I think this is an OpenFOAM issue. I also experienced reconstruct problems with a static mesh for large scale computations on a super computer. 

If you have an idea how to solve this, it would be greatly appreciated.

**Comment by FoamScience** [comment]
I see.

The problem seems to be originating from a bug in `reconstructPar`. It is somehow trying to read mesh data of the previous time steps instead of the current one!!!

There is a simple workaround. Reconstruction seems to work if you specify the time as a CLI argument:
```bash
reconstructParMesh
for t in $(foamListTimes); do reconstructPar -time $t; done
```

Or, even better, always reconstruct using `distributePar`:
```bash
runParallel redistributePar -parallel -reconstruct
```

I personally like the second way better. Thanks for reporting this behavior.

I'll keep this issue open; and track progress on fully resolving this bug. If I find time, I'll investigate this further during the weekend.

- [ ] add a wiki note about using `redistributePar`
- [ ] switch all tutorials to use `redistributePar` for decomposition/reconstruction
- [ ] fix `reconstructPar` bug: trying to read some processor meshes from the wrong time folders

**Comment by MahadevP2001** [comment]
After reconstructParMesh, when I run reconstructPar -time <timeValue> I sometimes get the warning : 

--> FOAM Warning :
    From virtual Foam::polyMesh::readUpdateState Foam::polyMesh::readUpdate()
    in file meshes/polyMesh/polyMeshIO.C at line 235
    Number of patches has changed.  This may have unexpected consequences.  Proceed with care.
--> FOAM Warning :
    From int main(int, char**)
    in file reconstructPar.C at line 369
    readUpdate for the reconstructed mesh:2
readUpdate for the processor meshes  :3
These should be equal or your addressing might be incorrect. Please check your time directories for any mesh directories.
Reconstructing FV fields

 Reconstructing volScalarFields

However , the reconstruction proceeds without exiting with a fatal error as described here. And I am able to visualize results on Paraview. Is the warning a cause for concern?

**Comment by FoamScience** [comment]
> After reconstructParMesh, when I run reconstructPar -time I sometimes get the warning :
> 
> --> FOAM Warning : From virtual Foam::polyMesh::readUpdateState Foam::polyMesh::readUpdate() in file meshes/polyMesh/polyMeshIO.C at line 235 Number of patches has changed. This may have unexpected consequences. Proceed with care. --> FOAM Warning : From int main(int, char**) in file reconstructPar.C at line 369 readUpdate for the reconstructed mesh:2 readUpdate for the processor meshes :3 These should be equal or your addressing might be incorrect. Please check your time directories for any mesh directories. Reconstructing FV fields
> 
> Reconstructing volScalarFields
> 
> However , the reconstruction proceeds without exiting with a fatal error as described here. And I am able to visualize results on Paraview. Is the warning a cause for concern?

Not really, I never got to the bottom of it, but I think that's just a warning the some boundary conditions may not support changing patches. Should be safe for all BCs that come with OpenFOAM; custom BCs shouldn't cache lists related to face/Cell IDs; If they do, they must provide functionality that maps cached lists (eg. autoMap methods)
